src/jit/valuenum.cpp

```diff
@@ -2809,14 +2810,11 @@ void Compiler::fgValueNumberArrIndexAssign(CORINFO_CLASS_HANDLE elemTypeEq,
         vnStore->vnDump(this, newValAtArrType);
         printf("\n");
 
-        printf("  fgCurHeapVN assigned:\n");
+        printf("  fgCurMemoryVN assigned:\n");
     }
 #endif // DEBUG
 
-    // bbHeapDef must be set to true for any block that Mutates the global Heap
-    assert(compCurBB->bbHeapDef);
-
-    fgCurHeapVN = vnStore->VNForMapStore(TYP_REF, fgCurHeapVN, elemTypeEqVN, newValAtArrType);
+    return vnStore->VNForMapStore(TYP_REF, fgCurMemoryVN[GcHeap], elemTypeEqVN, newValAtArrType);
```

Comment:
This code used to assign `fgCurHeapVN` but doesn't anymore.  Now it just returns the new value number, which then gets passed to `recordGcHeapStore` to do the update to `fgCurMemoryVN[GcHeap]`, and that function does have the updated assert.